### PR TITLE
Fix handling of URLs with special characters

### LINKED ISSUE #11: URLs with special characters don't work
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/bitbar/issues/11)

If a URL passed to `href` has special characters, such as single quotes, ampersands, etc., then the URL ends with that character when opened via this package to BitBar.

Per [this](https://stackoverflow.com/a/39707642/420217), seems like the single quote has to be specifically escaped after the `encodeURI`:

https://github.com/sindresorhus/bitbar/blob/4abea7e002a29fec7118af85119387a4530c7a76/index.js#L34

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

#
#
#
#
 
[
<
i
m
g
 
s
r
c
=
'
h
t
t
p
s
:
/
/
a
v
a
t
a
r
s
0
.
g
i
t
h
u
b
u
s
e
r
c
o
n
t
e
n
t
.
c
o
m
/
u
/
6
1
5
3
8
1
6
?
v
=
4
'
 
a
l
t
=
'
s
a
s
i
k
a
n
t
h
5
1
3
'
 
w
i
d
t
h
=
2
4
 
h
e
i
g
h
t
=
2
4
>
 
s
a
s
i
k
a
n
t
h
5
1
3
]
(
h
t
t
p
s
:
/
/
i
s
s
u
e
h
u
n
t
.
i
o
/
u
/
s
a
s
i
k
a
n
t
h
5
1
3
)
 
h
a
s
 
b
e
e
n
 
r
e
w
a
r
d
e
d
.

### Backers (Total: $20.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($20.00)
### Submitted pull Requests
- [#19 correctly encodes special characters in the `href` option](https://issuehunt.io/r/sindresorhus/bitbar/pull/19)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/bitbar/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by swarnicapankaj**
Fixes #11 
Added a new test to test this fix

The test passed.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

**Comment by sindresorhus** [comment]
Have you tested it with the actual Bitbar app that it works in practice?

**Comment by swarnicapankaj** [comment]
Yes I verified by adding a plugin which uses a quote in the url. The quote gets encoded successfully.

File: test.js [hunk 1]
```diff
@@ -51,6 +51,56 @@ Ponies|
 	t.is(actual, expected);
 });
 
+test('main2', t => {
```
Comment (sindresorhus) [inline test.js]: This needs a better name.

File: test.js [hunk 2]
```diff
@@ -51,6 +51,56 @@ Ponies|
...
+	const actual = bitbar.create([
```
Comment (sindresorhus) [inline test.js:55]: This test has too much boilerplate. You only need to test with one item having an URL with a quote.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: test.js (hunk 1 above)
Comment (swarnicapankaj) [reply test.js]: Renamed to `test_quotes_encoding_in_href`

File: test.js (hunk 2 above)
Comment (swarnicapankaj) [reply test.js:55]: Removed unnecessary codes.

**Comment by swarnicapankaj** [comment]
Test passed:
```
bash-3.2$ npx ava -v

  ✔ main
  ✔ test_quotes_encoding_in_href
  ✔ `text` property validation

  3 tests passed

```

File: test.js [hunk 3]
```diff
@@ -51,6 +51,22 @@ Ponies|
 	t.is(actual, expected);
 });
 
+test('correctly encodes quotes in the `href` option', t => {
+	const actual = bitbar.create([
+		{
+			text: 'Unicorns',
+			color: '#ff79d7',
+			href: 'https://www.youtube.com/watch?v=\'9auOCbH5Ns4&time=2',
```
Comment (sindresorhus) [inline test.js:59]: #11 also mentions ampersands, can you include that in the test here too? And also test manually with BitBar? I would also include double-quotes, just to be sure.

**Comment by sindresorhus** [comment]
Ping

**Pull request closed without merging**